### PR TITLE
[Checkbox#329] Add new intents

### PR DESCRIPTION
**Pull request by aycil-alican**
### Pull requests template

This pull request has:
- [ ] Code documentation on publics
- [ ] Accessibility
	- [ ] Dynamic sizes
	- [ ] Identifiers
- [ ] Tests
- [ ] Demo update
- [ ] Wiki

### CODE REVIEW

File: core/Sources/Components/Checkbox/Enum/CheckboxState.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//  CheckboxState.swift
+//  SparkCore
+//
+//  Created by janniklas.freundt.ext on 11.04.23.
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+import Foundation
+
+public enum CheckboxState: CaseIterable {
```
Comment (michael-zimmermann) [inline core/Sources/Components/Checkbox/Enum/CheckboxState.swift:11]: To be honest, I'd deprecate this and just work with a boolean as in most UIKit and SwiftUI controls.

File: core/Sources/Components/Checkbox/UseCase/Colors/CheckboxColorsUseCase.swift [hunk 2]
```diff
@@ -7,31 +7,86 @@
 //
 
 protocol CheckboxColorsUseCaseable {
-    func execute(from theme: Theme, state: SelectButtonState) -> CheckboxColorables
+    func execute(from color: Colors, intent: CheckboxIntent) -> CheckboxColors
```
Comment (michael-zimmermann) [inline core/Sources/Components/Checkbox/UseCase/Colors/CheckboxColorsUseCase.swift:10]: 👍

File: core/Sources/Components/Checkbox/View/CheckboxGroupItemProtocol.swift [hunk 3]
```diff
@@ -23,5 +23,5 @@ public protocol CheckboxGroupItemProtocol: Hashable {
     var selectionState: CheckboxSelectionState { get set }
 
     /// The current control state of the checkbox.
-    var state: SelectButtonState { get set }
```
Comment (michael-zimmermann) [inline core/Sources/Components/Checkbox/View/CheckboxGroupItemProtocol.swift:26]: Maybe the old SelectButtonState needs to be kept and made deprecated. It is already used in SparkKit on Polaris, and I'm not sure if only the `enabled` state is used.

**Comment by aycil-alican** [comment]
This PR will merge with this https://github.com/adevinta/spark-ios/pull/482

**Pull request closed without merging**